rules/no-await-in-promise-methods.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+const isPromiseMethodWithArray = require('./utils/is-promise-method-with-array.js');
+
+const MESSAGE_ID = 'no-await-in-promise-methods';
+const messages = {
+	[MESSAGE_ID]: 'Parameters in `Promise.{{method}}` should not be awaited.',
+};
+
+const getArrayElements = node => node.arguments[0].elements;
+
+const isAwait = element => element.type === 'AwaitExpression';
+
+const getMethodName = node => node.callee.property.name;
+
+const getFixer = ({sourceCode}, element) => fixer =>
+	fixer.replaceText(element, sourceCode.getText(element.argument));
+
+/** @param {import('eslint').Rule.RuleContext} context */
+const create = context => ({
+	CallExpression(node) {
+		if (isPromiseMethodWithArray(node)) {
+			for (const element of getArrayElements(node)) {
+				if (isAwait(element)) {
```

Comment:
Please use `element.type === 'AwaitExpression'`, there can also be holes in array, 


```js
Promise.all([/* a hole */,])
```

Reply:
It's corrected.